test/java/seedu/addressbook/logic/LogicTest.java

```diff
@@ -92,22 +92,48 @@ private void assertCommandBehavior(String inputCommand,
         assertEquals(lastShownList, logic.getLastShownList());
         assertEquals(addressBook, saveFile.load());
     }
+    /**
+     * Executes the command and confirms that the result message is correct and
+     * also confirms that the following three parts of the Logic object's state are as expected:<br>
+     *      - the internal address book data are same as those in the {@code expectedAddressBook} <br>
+     *      - the internal 'last shown list' matches the {@code expectedLastList} <br>
+     */
+    private void assertCommandBehaviorWithoutSave(String inputCommand,
```

Comment:
Instead of doing this, you can save a lot of code by using the original assert method but check the mutability of the command inside this method. You can refer your peers' code on this if you are not sure. Good effort anyways!